lib/facter/java_version.rb

```diff
@@ -11,6 +11,11 @@
 #
 # Notes:
 #   None
+osfamily = Facter.value(:osfamily)
+case osfamily
+when 'OpenBSD'
+  ENV['PATH'] = ENV['PATH'] + File::PATH_SEPARATOR + '/usr/local/jdk-1.7.0/jre/bin'
```

Comment:
and while we're at it, and java 7 will be EOL rsn, you could replace that 7 with 8…